src/shared/mount-util.c

```diff
@@ -489,6 +490,52 @@ int mount_move_root(const char *path) {
         return RET_NERRNO(chdir("/"));
 }
 
+int mount_pivot_root(const char *path) {
+        _cleanup_close_ int fd_oldroot = -EBADF, fd_newroot = -EBADF;
+
+        assert(path);
+
+        /* pivot_root() isn't currently supported in the initramfs. */
+        if (in_initrd())
+                return mount_move_root(path);
+
+        fd_oldroot = open("/", O_PATH|O_DIRECTORY|O_CLOEXEC|O_NOFOLLOW);
+        if (fd_oldroot < 0)
+                return log_debug_errno(errno, "Failed to open old rootfs");
+
+        fd_newroot = open(path, O_PATH|O_DIRECTORY|O_CLOEXEC|O_NOFOLLOW);
+        if (fd_newroot < 0)
+                return log_debug_errno(errno, "Failed to open new rootfs '%s': %m", path);
+
+        /* Change into the new rootfs. */
+        if (fchdir(fd_newroot) < 0)
+                return log_debug_errno(errno, "Failed to change into new rootfs '%s': %m", path);
+
+        /* Let the kernel tuck the new root under the old one. */
+        if (pivot_root(".", ".") < 0)
+                return log_debug_errno(errno, "Failed to pivot root to new rootfs '%s': %m", path);
+
+
```

Comment:
Doubled empty line.

Reply:
This is fixed in another branch of mine luckily.